lib/checks/canFindRoots.js

```diff
@@ -0,0 +1,39 @@
+const Node = require('../node');
+const resolvesToConstant = require('./resolvesToConstant.js');
+
+function canFindRoots(equation) {
+
+  // Equation should be in the form of factor * factor = 0, factor^power = 0, or identifier = 0
+  // e.g (x - 2)^2 = 0, x(x + 2)(x - 2) = 0
+  const left = equation.leftNode;
+  const right = equation.rightNode;
+
+  const zeroRightSide = Node.Type.isConstant(right)
+        && parseFloat(right.value) === 0;
+
+  const isMulOrPower = Node.Type.isOperator(left, '*') || Node.Type.isOperator(left, '^');
+
+  if (!(zeroRightSide && isMulOrPower)) {
+    return false;
+  }
+
+  // If the left side of the equation is multiplication, filter out all the factors
+  // that do not evaluate to constants because they do not have roots. If the
```

Comment:
This comment is confusing me -- do you mean "filter out the factors that _do_ evaluate to constants"?